### PR TITLE
[MI-2406] Add a menu in rhs

### PR DESCRIPTION
**Pull request by ayusht2810**
image

### CODE REVIEW

File: webapp/src/containers/Rhs/rhsHeader.tsx [hunk 1]
```diff
@@ -58,22 +60,20 @@ const Header = ({
...
-                            >
-                                {SVGIcons.share}
-                            </SvgWrapper>
-                        </span>
-                        {Constants.ShareRecordButton}
-                    </button>
+                        <SvgWrapper
+                            width={16}
+                            height={28}
+                            viewBox='0 -1 11 10'
```
Comment (manojmalik20) [inline webapp/src/containers/Rhs/rhsHeader.tsx:71]: Also, get this PR reviewed by a FE dev.

File: webapp/src/styles/utils/_utils.scss [hunk 2]
```diff
@@ -41,3 +41,8 @@
 .height-290 {
     height: 290px;
 }
+
+// Padding
+.padding-left-6 {
```
Comment (manojmalik20) [inline webapp/src/styles/utils/_utils.scss:46]: I think these should be dynamic. I mean whatever value we provide after `padding-left-`, that should be a variable. Check with a FE dev and see if it is too complex.

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/containers/Rhs/__snapshots__/rhsHeader.test.tsx.snap [hunk 3]
```diff
@@ -29,25 +29,23 @@ exports[`RHSHeader should match snapshot 1`] = `
           />
         </SVGWrapper>
       </IconButton>
-      <button
-        className="btn btn-primary share-record-btn false"
+      <IconButton
+        extraClass="false"
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/__snapshots__/rhsHeader.test.tsx.snap:33]: can't we just not pass this prop if no `extraClass` needed? what is `extraClass="false"`?

File: webapp/src/containers/Rhs/rhs.scss [hunk 4]
```diff
@@ -81,6 +81,12 @@
     box-shadow: 0 8px 24px rgba(0, 0, 0, 0.15);
   }
 
+  .rhs-menu-popover {
+    right: 12px;
+    width: 310px;
+    padding: 0px;
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/rhs.scss:87]:
```suggestion
    padding: 0;
```
no need to specify unit for `0`

File: webapp/src/containers/Rhs/rhsHeader.tsx [hunk 5]
```diff
@@ -58,22 +60,20 @@ const Header = ({
                             </SvgWrapper>
                         </IconButton>
                     )}
-                    <button
-                        className={`btn btn-primary share-record-btn ${!showFilterIcon && 'share-record-btn__without-filter'}`}
-                        onClick={() => dispatch(setGlobalModalState({modalId: 'shareRecord'}))}
+                    <IconButton
+                        tooltipText='Actions'
+                        onClick={() => setShowMenu(!showMenu)}
+                        extraClass={`${!showFilterIcon && 'margin-left-auto'}`}
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/rhsHeader.tsx:66]:
```suggestion
                        extraClass={!showFilterIcon && 'margin-left-auto'}
```

File: webapp/src/styles/utils/_utils.scss (hunk 2 above)
Comment (avas27JTG) [reply webapp/src/styles/utils/_utils.scss:46]: that is implemented in UI Lib, here I don't think we need too many of such util classes so for 4-5 such classes we can write it this way, but if such utils are frequently used then we can take reference from UI Lib and implement that logic here in `SCSS`

Comment (ayusht2810) [reply webapp/src/styles/utils/_utils.scss:46]: discussed with @manojmalik20, we are not implementing this

File: webapp/src/containers/Rhs/__snapshots__/rhsHeader.test.tsx.snap (hunk 3 above)
Comment (ayusht2810) [reply webapp/src/containers/Rhs/__snapshots__/rhsHeader.test.tsx.snap:33]: this is a snapshot

File: webapp/src/containers/Rhs/rhsHeader.tsx (hunk 5 above)
Comment (ayusht2810) [reply webapp/src/containers/Rhs/rhsHeader.tsx:66]: getting typescript error on doing this

Comment (avas27JTG) [reply webapp/src/containers/Rhs/rhsHeader.tsx:66]: try `extraClass={!showFilterIcon ? 'margin-left-auto' : ''}`

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**